cEP-0020.md

```diff
@@ -0,0 +1,418 @@
+
+## Abstract
+
+This cEP describes how a gamification system will automate the process
+of becoming a `Developer` from a `Newcomer`, as a part of this
+[GSoC project](https://summerofcode.withgoogle.com/projects/#5892040252981248).
+
+For the implementation of this process, This project will incorporate a
+gamification system which will allow us to track newcomer's progress and
+assign some points to each of the activity they do. There will be different levels
+based on points threshold and a newcomer will also get some special privileges
+after the completion of each level. There will also be different badges on the
+basis of performed activities by the newcomers.
+
+So, every newcomer at coala will have to complete the processes involved in
+the gamification system in order to become a developer at coala or to get the
+**coala Developer** badge.
+
+## Background
+
+### Newcomer Experience
+
+**The problems we have:**
+
+- How to attract more newcomers
+  and how to cope with the increasing amounts of newcomers
+- How to teach and help newcomers to learn faster so
+  we can keep working on coala at a high pace.
+- How to improve newcomer -> dev/maintainer ratio
+- How to increase amount of feedback we get
+  and how to use those feedback to improve coala and its processes
+
+## Why Gamification?
+
+As coala tries to be a welcoming organization for newcomers, it gives them
+clear pathways and as much direct assistance as possible. However, it is
+already known that newcomers face many barriers while attempting to contribute
+to open source for the first time, and additional barriers while attempting
+to contribute to projects run by the coala organization.
+
+Some barriers they face include orientation issues that can potentially
+demotivate newcomers from placing their first contribution. On the other
+hand, gamification is widely used to engage and motivate people to
+accomplish tasks and improve their performance.
+
+Therefore, the goal of this project is to use gamification to orient and
+motivate newcomers to overcome onboarding barriers to contribute and engage
+with coala and its community.
+
+## Gamification Introduction
+
+There will be four game design elements to help newcomers: Activities, Points,
+Badges and Levels. For each of these elements, we will have a set of rules to
+describe the operation of the gamified environment. i.e.:
+
+### Activities
+
+Let's start by defining some activities which could be tracked:
+
+- Create a bug report
+- Create newcomer difficulty issue
+- Create low difficulty issue
+- Create medium difficulty issue
+- Create a bear proposal issue
+- Create pull request that fixes a bug
+- Create pull request for bear proposal
+- Create pull request for newcomer issue
+- Create pull request for low difficulty issue
+- Create pull request for medium difficulty issue
+- Gitmate errors on pull request
+- CI status of the pull request
+- Review comments on pull request
+- Messages on Gitter channels
```

Comment:
all of the above are complete defined activities, but here you group them, and group them very broadly into good / bad.

At least get rid of this grouping, and use "Send a good message to coala community on Gitter" and the same for bad message.

But I think it would be better to break this down into discrete useful activities, like 'Say "hello world" on Gitter' , "asking to be assigned to an issue on gitter" , using coala gitter rooms other than the main "coala/coala".

You can also add "use corobo on Gitter (currently disabled due to ....)"